### PR TITLE
Api spec sprint18 service settings

### PR DESCRIPTION
**Pull request by jenjenp**

### CODE REVIEW

File: services.yaml [hunk 1]
```diff
@@ -174,12 +176,32 @@ paths:
           $ref: "#/responses/TBD"
         401:
           $ref: "#/responses/UnauthorizedError"
+  /services/{service_id}/rename:
```
Comment (chook) [inline services.yaml]: why do we need a POST for /rename? i think a POST on /{service_id} with name="new name" has a similar affect and is more general

File: services.yaml [hunk 2]
```diff
@@ -199,9 +221,30 @@ paths:
...
       tags:
-        - General
-      summary: "Add members to a service key"
-      description: "Add members to a service key"
+        - Team Management
+      summary: "Add members to an environment"
+      description: "Add new members to the environment"
+      parameters:
+          - in: path
+            name: service_id
```
Comment (chook) [inline services.yaml]: once you merge master to this, make sure to change these service_id to env_id

File: services.yaml [hunk 3]
```diff
@@ -218,6 +261,37 @@ paths:
           $ref: "#/responses/OK"
         401:
           $ref: "#/responses/UnauthorizedError"
+  /services/{service_id}/team/change-role:
```
Comment (chook) [inline services.yaml]: this to me feels like we are changing the role of the team and not members of it.
i think a POST team/{member} is nicer.
If we are in need of batch here, than i would simply make the /team endpoint more sophisticated (and less restful) and have it as a general invite-or-change-member

**Review by chook — CHANGES_REQUESTED** [review]

File: services.yaml [hunk 4]
```diff
@@ -2978,6 +3052,10 @@ definitions:
       full_key:
         type: string
         example: S35817XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+      role:
```
Comment (chook) [inline services.yaml:3182]: this is no longer a ServiceBasic as a Service shouldn't be coupled with a "user" in such a way.
We should create a new item that has both a ServiceBasic and a role as children

File: services.yaml (hunk 1 above)
Comment (jenjenp) [reply services.yaml]: you mean using the regular edit command and the full service object for renaming?

File: services.yaml (hunk 2 above)
Comment (jenjenp) [reply services.yaml]: sure, will do

File: services.yaml (hunk 3 above)
Comment (jenjenp) [reply services.yaml]: what about change-user-role?

**Review by jenjenp — COMMENTED** [review]
So, do you think we shall add a different endpoint to fetch the user role?

File: services.yaml [hunk 5]
```diff
@@ -3172,7 +3140,29 @@ definitions:
       initiator_type:
         type: string
         example: "User"
-  ServiceBasic:
+  EnvironmentWithRole:
+    type: object
+    description: Extended information about a environment key
```
Comment (chook) [inline services.yaml]: an environment, an no need for "key"

File: services.yaml [hunk 6]
```diff
@@ -3172,7 +3140,29 @@ definitions:
...
+    properties:
+      EnvironmentBasic:
```
Comment (chook) [inline services.yaml]: i'm not sure this does what you mean, you need a $ref
look at Event / EventStats

File: services.yaml [hunk 7]
```diff
@@ -3184,12 +3174,21 @@ definitions:
...
         example: S35817XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
-      role:
-        type: string
-        description: user's role in the environment
-        example: admin
-  TeamMembers:
-    description: List of environment key team members
+  TeamMembersBasic:
+    description: List of environment team members
+    type: array
```
Comment (chook) [inline services.yaml:3201]: here as well, i would add some object composition

File: services.yaml (hunk 5 above)
Comment (jenjenp) [reply services.yaml]: fixed to: Environment data including user's role

File: services.yaml (hunk 7 above)
Comment (jenjenp) [reply services.yaml:3201]: In this case it's similar to viewfull and viewbasic - teambasic includes fields editable by admins, teamfull includes all generated and personal data that can't be edited by admin

File: services.yaml (hunk 6 above)
Comment (jenjenp) [reply services.yaml]: didn't work in spectacle

Comment (jenjenp) [reply services.yaml]: Let's start with this one and when additional option will be added to ServiceBasic, will change to #ref

File: services.yaml [hunk 8]
```diff
@@ -230,8 +230,8 @@ paths:
     post:
       tags:
         - Team Management
-      summary: "Add members to an environment"
-      description: "Add new members to an environment"
+      summary: "Add new members"
+      description: "Add new members"
```
Comment (chook) [inline services.yaml:234]: longer description (not blocking)

File: services.yaml [hunk 9]
```diff
@@ -269,6 +269,28 @@ paths:
           $ref: "#/responses/OK"
         401:
           $ref: "#/responses/UnauthorizedError"
+  /services/{env_id}/team/change-role:
+    post:
+      tags:
+        - Team Management
+      summary: "Edit roles"
```
Comment (chook) [inline services.yaml:276]: if the URL is change-role, why the summary is "edit roles"? i would stick to one verb (not blocking)

**Pull request merged**